### PR TITLE
prefer endians2

### PR DESCRIPTION
**Pull request by arnetheduck**

### CODE REVIEW

File: beacon_chain/spec/helpers.nim [hunk 1]
```diff
@@ -91,12 +92,12 @@ func bytes_to_int*(data: openarray[byte]): uint64 =
 func int_to_bytes32*(x: uint64): array[32, byte] =
   ## Little-endian data representation
   ## TODO remove uint64 when those callers fade away
-  littleEndian64(result[0].addr, x.unsafeAddr)
+  result[0..<7] = x.toBytesLE()
 
 func int_to_bytes32*(x: Epoch): array[32, byte] {.borrow.}
 
 func int_to_bytes8*(x: uint64): array[8, byte] =
```
Comment (zah) [inline beacon_chain/spec/helpers.nim]: This function became too simple. There is no much value in keeping it around

File: beacon_chain/ssz.nim [hunk 2]
```diff
@@ -117,20 +117,9 @@ func writeFixedSized(c: var WriteCursor, x: auto) =
...
-      elif x.sizeof == 4: littleEndian32(addr bytes[0], x.unsafeAddr)
-      elif x.sizeof == 2: littleEndian16(addr bytes[0], x.unsafeAddr)
-      elif x.sizeof == 1: copyMem(addr bytes[0], x.unsafeAddr, sizeof(x))
-      else: unsupported x.type
-      c.append bytes
-    else:
-      let valueAddr {.used.} = unsafeAddr x
-      trs "APPENDING INT ", x, " = ", makeOpenArray(cast[ptr byte](valueAddr), sizeof(x))
-      c.appendMemCopy x
+    let value = x.toBytesLE()
```
Comment (zah) [inline beacon_chain/ssz.nim]: I'm not sure that the compiler will be able to optimize away this copy of the bytes of the integer. The previous version of the code did not introduce such a copy.

Comment (arnetheduck) [reply beacon_chain/ssz.nim]: hm, I'd be a bit surprised if it didn't - I mean, it optimizes away a lot more advanced code that @cheatfate was concerned about: https://gcc.godbolt.org/z/lACv_t - in particular relevance to this case, it optimizes away the loop that makes the endian conversion alignment-safe, so I'm hoping this assignment would go away as well.

I can add back the when stuff, I just tend to be reluctant to do so without a benchmark proving its usefulness - it is fairly ugly after all - up to you.

File: beacon_chain/spec/helpers.nim (hunk 1 above)
Comment (arnetheduck) [reply beacon_chain/spec/helpers.nim]: it's a spec function, else I would have removed it indeed:  https://github.com/ethereum/eth2.0-specs/blob/v0.10.1/specs/phase0/beacon-chain.md#int_to_bytes

File: nfuzz/libnfuzz.nim [hunk 3]
```diff
@@ -57,195 +56,77 @@ proc copyState(state: BeaconState, output: ptr byte,
...
-    raise newException(
-      FuzzCrashError,
-      "Unexpected (logging?) error in deposit processing",
-      e,
-    )
-
-  if result:
-    result = copyState(data.state, output, output_size)
+  decodeAndProcess(DepositInput):
+    process_deposit(data.state, data.deposit, flags)
```
Comment (zah) [inline nfuzz/libnfuzz.nim]: You've removed the different strings associated with the `FuzzCrashError` exceptions. Not sure how important they are for @gnattishness's crash reports.

Comment (zah) [reply nfuzz/libnfuzz.nim]: Nevermind, I missed some of the code, sorry

**Pull request merged**